### PR TITLE
Add favourites feature and update description to fit project

### PR DESCRIPTION
**Pull request by takufunkai**
Fixes 2/6 of #22.
- Updates the description to refer to our project instead of `AddressBook`
- Adds `Favourites` User Guide feature written for tutorial 
- [x] TODO: update list of commands table (cr: @glennljw)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -30,11 +30,15 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
 
    * **`delete`**`3` : Deletes the 3rd contact shown in the current list.
 
+   * **`favourite`**`2` : Adds the 2nd contact shown in the current list to your list of favourite contacts
+   
+   * **`favourite`** : Lists all your favourite contacts.
```
Comment (weijuey) [inline docs/UserGuide.md:35]: I believe the command word for this is `favourites` ?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -107,8 +111,26 @@ Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
...
 
 Examples:
-*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
-*  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+* `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
+* `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+
+### Adding favourites: `favourite`
+
+Adds the client to your list of favourite clients and view them.
```
Comment (weijuey) [inline docs/UserGuide.md:119]: Something I encountered in transferring user stories into the DG is that we might be using the terms "person", "contact" and "client" too interchangeably, which might be confusing. I suggest standardising the term for an entry in the app to be "contact", would that be alright?

**Review by weijuey — CHANGES_REQUESTED** [review]
Lgtm, just some nits to fix. :+1:

File: docs/UserGuide.md (hunk 2 above)
Comment (takufunkai) [reply docs/UserGuide.md:119]: Ok!

Comment (weijuey) [reply docs/UserGuide.md:119]: PS, didn't see the Draft initially.

Comment (takufunkai) [reply docs/UserGuide.md:119]: Going to leave this for now in this PR, will change it all in one shot in another PR

**Review by weijuey — APPROVED** [review]
Lgtm :+1:

**Pull request merged**